### PR TITLE
chain: Implement do-not-modify for basic strings

### PR DESCRIPTION
**Pull request by sorawit**

### CODE REVIEW

**Review by Benzbeeb — DISMISSED** [review]

File: chain/x/oracle/keeper/data_source_test.go [hunk 1]
```diff
@@ -75,6 +75,36 @@ func TestAddDataSourceEditDataSourceBasic(t *testing.T) {
...
+	_, ctx, k := createTestInput()
+	// Creates some basic data sources.
+	dataSource1 := types.NewDataSource(
+		Alice.Address, "NAME1", "DESCRIPTION1", []byte("executable1"),
+	)
+	dataSource2 := types.NewDataSource(
+		Bob.Address, types.DoNotModify, types.DoNotModify, []byte("executable2"),
+	)
+	// Adds a new data source to the store. We should be able to retreive it back.
+	id, err := k.AddDataSource(ctx, types.NewDataSource(
```
Comment (taobun) [inline chain/x/oracle/keeper/data_source_test.go]: Why don't write `k.AddDataSource(ctx, dataSource1)`?

File: chain/x/oracle/keeper/data_source_test.go [hunk 2]
```diff
@@ -75,6 +75,36 @@ func TestAddDataSourceEditDataSourceBasic(t *testing.T) {
...
+		dataSource1.Owner, dataSource1.Name, dataSource1.Description, dataSource1.Executable,
+	))
+	require.Nil(t, err)
+	require.Equal(t, dataSource1, k.MustGetDataSource(ctx, id))
+	require.NotEqual(t, dataSource2, k.MustGetDataSource(ctx, id))
+	// Edits the data source. We should get the updated data source.
+	err = k.EditDataSource(ctx, id, types.NewDataSource(
```
Comment (taobun) [inline chain/x/oracle/keeper/data_source_test.go]: ditto

File: chain/x/oracle/keeper/oracle_script_test.go [hunk 3]
```diff
@@ -77,6 +77,41 @@ func TestAddEditOracleScriptBasic(t *testing.T) {
...
+	// Creates some basic oracle scripts.
+	oracleScript1 := types.NewOracleScript(
+		Alice.Address, "NAME1", "DESCRIPTION1", []byte("code1"), BasicSchema, BasicSourceCodeURL,
+	)
+	oracleScript2 := types.NewOracleScript(
+		Bob.Address, types.DoNotModify, types.DoNotModify, []byte("code2"),
+		types.DoNotModify, types.DoNotModify,
+	)
+	// Adds a new oracle script to the store. We should be able to retreive it back.
+	id, err := k.AddOracleScript(ctx, types.NewOracleScript(
```
Comment (taobun) [inline chain/x/oracle/keeper/oracle_script_test.go]: ditto

File: chain/x/oracle/keeper/oracle_script_test.go [hunk 4]
```diff
@@ -77,6 +77,41 @@ func TestAddEditOracleScriptBasic(t *testing.T) {
...
+		oracleScript1.Owner, oracleScript1.Name, oracleScript1.Description, oracleScript1.Code,
+		oracleScript1.Schema, oracleScript1.SourceCodeURL,
+	))
+	require.Nil(t, err)
+	require.Equal(t, oracleScript1, k.MustGetOracleScript(ctx, id))
+	require.NotEqual(t, oracleScript2, k.MustGetOracleScript(ctx, id))
+	// Edits the oracle script. We should get the updated oracle script.
+	err = k.EditOracleScript(ctx, id, types.NewOracleScript(
```
Comment (taobun) [inline chain/x/oracle/keeper/oracle_script_test.go]: ditto

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — DISMISSED** [review]

**Pull request merged**